nilearn/_utils/path_finding.py

```diff
@@ -10,7 +10,7 @@ def resolve_globbing(path):
     """Resolve globbing patterns in a path."""
     path = stringify_path(path)
     if isinstance(path, str):
-        path_list = sorted(glob.glob(os.path.expanduser(path)))
+        path_list = sorted(glob.glob(str(Path(path).expanduser())))
```

Comment:
probably better to refactor this whole function in another PR to only use Path to avoid doing things like `str(Path("foo"))` just to silence the ruff error.